modules/navigation/navigation_mesh_generator.cpp

```diff
@@ -631,7 +631,12 @@ void NavigationMeshGenerator::bake(Ref<NavigationMesh> p_nav_mesh, Node *p_node)
 
 #ifdef TOOLS_ENABLED
 	EditorProgress *ep(nullptr);
-	if (Engine::get_singleton()->is_editor_hint()) {
+	// FIXME
+	// After discussion on devchat disabled EditorProgress for now as it is not thread-safe and uses hacks and Main::iteration() for steps.
+	// EditorProgress randomly crashes the Engine when the bake function is used with a thread e.g. inside Editor with a tool script and procedural navigation
+	// This was not a problem in older versions as previously Godot was unable to (re)bake NavigationMesh at runtime.
+	// If EditorProgress is fixed and made thread-safe this should be enabled again.
+	if (Engine::get_singleton()->is_editor_hint() && false) {
```

Comment:
Instead of adding `false` (some compilers might raise a warning that it's a tautological condition) I'd move the comment before the `#ifdef` and change it to a `#if 0`, so you skip the whole EP specific code.